service/controller/cert.go

```diff
@@ -228,20 +227,6 @@ func tenantClusterExists(k8sClient k8sclient.Interface, id string) (bool, error)
 		}
 	}
 
-	// We need to check for the legacy AzureConfig CRs on Azure environments.
-	{
```

Comment:
We use CAPI `Cluster` CRD on `azure` so there is no need to look for `AzureConfigs`